web/main.go

```diff
@@ -27,6 +27,7 @@ func main() {
 	staticDir := flag.String("static-dir", "app/dist", "directory to search for static files")
 	reload := flag.Bool("reload", true, "reloading set to true or false")
 	controllerNamespace := flag.String("controller-namespace", "linkerd", "namespace in which Linkerd is installed")
+	clusterDomain := flag.String("cluster-domain", "cluster.local", "custom cluster domain")
```

Comment:
Similar to `grpc_server.go`, we should be able to read this from the `linkerd-config` config map. Using the code snippet from https://github.com/linkerd/linkerd2/pull/2930, e.g.

```diff
diff --git a/web/srv/handlers.go b/web/srv/handlers.go
index f0443102..4d695b53 100644
--- a/web/srv/handlers.go
+++ b/web/srv/handlers.go
@@ -9,6 +9,8 @@ import (
 	"github.com/julienschmidt/httprouter"
 	"github.com/linkerd/linkerd2/controller/api/public"
 	pb "github.com/linkerd/linkerd2/controller/gen/public"
+	"github.com/linkerd/linkerd2/pkg/config"
+	pkgK8s "github.com/linkerd/linkerd2/pkg/k8s"
 	profiles "github.com/linkerd/linkerd2/pkg/profiles"
 	log "github.com/sirupsen/logrus"
 )
@@ -69,7 +71,13 @@ func (h *handler) handleProfileDownload(w http.ResponseWriter, req *http.Request
 	}
 
 	profileYaml := &bytes.Buffer{}
-	err := profiles.RenderProfileTemplate(namespace, service, profileYaml)
+
+	globalConfig, err := config.Global(pkgK8s.MountPathGlobalConfig)
+	if err != nil {
+	}
+	clusterDomain := globalConfig.GetClusterDomain()
+
+	err = profiles.RenderProfileTemplate(namespace, service, clusterDomain, profileYaml)
 
 	if err != nil {
 		log.Error(err)
```

Reply:
Same as above. I'll try to redo this part.

Thanks for the snippet!

Reply:
Removed from this PR. Will be readded in the next.